### PR TITLE
Add appendix with table of node types

### LINKED ISSUE #165: Add appendix listing all node types and edge constraints
There are some node types currently not included in it:

- `Const`
- `Call`
- `LoadConst`
- `Declare`
- `AlisasDeclare`
- `AliasDef`

(Also `CallIndirect` but this should probably move to the Tierkreis/higher-order resource; see #154 .)

### PR DESCRIPTION
**Pull request by cqc-alec**
Closes #165 .

### CODE REVIEW

File: specification/hugr.md [hunk 1]
```diff
@@ -1915,3 +1917,35 @@ differences include:
...
+### Appendix 2: Node types and their edges
+
+The following table shows which edge kinds may adjoin each node type.
+
+Under each edge kind, the inbound constraints are followed by the outbound
+constraints. The symbol ✱ atands for "any number". For example, "1, ✱" means
+"one edge in, any number out".
+
+Except for `Module` it is assumed that these are non-root nodes. The `Module`
+row of the table applies to root nodes of any type.
```
Comment (ss2165) [inline specification/hugr.md:1930]: Could just call it "Root" then?

File: specification/hugr.md [hunk 2]
```diff
@@ -1915,3 +1917,35 @@ differences include:
...
+
+| Node type      | `Value` | `Order` | `Static` | `ControlFlow` | `Hierarchy` | Children |
+| -------------- | ------- | ------- |--------- | ------------- | ----------- | -------- | 
+| `Module`       | 0, 0    | 0, 0    | 0, 0     | 0, 0          | 0, ✱        |          |
+| `Def`          | 0, 0    | 0, 0    | 0, ✱     | 0, 0          | 1, ✱        | DSG      |
+| `Declare`      | 0, 0    | 0, 0    | 0, ✱     | 0, 0          | 1, 0        |          |
+| `AliasDef`     | 0, 0    | 0, 0    | 0, 0     | 0, 0          | 1, 0        |          |
+| `AliasDeclare` | 0, 0    | 0, 0    | 0, 0     | 0, 0          | 1, 0        |          |
+| `Const`        | 0, 0    | 0, 0    | 0, ✱     | 0, 0          | 1, 0        |          |
+| `LoadConst`    | 0, 1    | 1, 1    | 1, 0     | 0, 0          | 1, 0        |          |
```
Comment (ss2165) [inline specification/hugr.md:1940]: Could introduce "+" to mean "1 or more" and use it for here? And outgoing order edges are any number right?
```suggestion
| `LoadConst`    | 0, 1    | 1, *    | 1, 0     | 0, 0          | 1, 0        |          |
```

**Review by ss2165 — APPROVED** [review]
This is a really useful reference

File: specification/hugr.md (hunk 2 above)
Comment (cqc-alec) [reply specification/hugr.md:1940]: and 1 out to the Output

Comment (cqc-alec) [reply specification/hugr.md:1940]: OK

**Review by ss2165 — APPROVED** [review]

**Pull request merged**